src/utils.rs

```diff
@@ -234,6 +234,78 @@ pub(crate) fn id_to_path(id: &str) -> String {
         .replace(".md", std::path::MAIN_SEPARATOR.to_string().as_str())
 }
 
+/// returns true if an existing file named "file_name"
+/// exists in the root package folder
+fn is_file_in_root(root: &str, file_name: &str) -> bool {
+    camino::Utf8PathBuf::from(root).join(file_name).exists()
+}
+
+/// returns favicon html tag as string
+/// (if favicon is passed as header in fpm.package or if any favicon.* file is present in the root package folder)
+/// otherwise returns None
+fn resolve_favicon(
+    root_path: &str,
+    package_name: &str,
+    favicon: &Option<String>,
+) -> Option<String> {
+    /// returns html tag for using favicon.
+    fn favicon_html(favicon_path: &str, content_type: &str) -> String {
+        let favicon_html = format!(
+            "\n<link rel=\"shortcut icon\" href=\"{}\" type=\"{}\">",
+            favicon_path, content_type
+        );
+        favicon_html
+    }
+
+    /// returns relative favicon path from package and its mime content type
+    fn get_favicon_path_and_type(package_name: &str, favicon_path: &str) -> (String, String) {
+        // relative favicon path wrt package
+        let path = camino::Utf8PathBuf::from(package_name).join(favicon_path);
+        // mime content type of the favicon
+        let content_type = mime_guess::from_path(path.as_str()).first_or_octet_stream();
+
+        (path.to_string(), content_type.to_string())
+    }
+
+    // favicon image path from fpm.package if provided
+    let fav_path = favicon;
+
+    let (full_fav_path, fav_mime_content_type): (String, String) = {
+        match fav_path {
+            Some(ref path) => {
+                // In this case, favicon is provided with fpm.package in FPM.ftd
+                get_favicon_path_and_type(package_name, path)
+            }
+            None => {
+                // If favicon not provided so we will look for favicon in the package directory
+                // By default if any file favicon.* is present we will use that file instead
+                // In case of favicon.* conflict priority will be: .ico > .svg > .png > .jpg.
+                // Default searching directory being the root folder of the package
+
+                // Just check if any favicon exists in the root package directory
+                // in the above mentioned priority order
+                let found_favicon_id: &str;
+                if is_file_in_root(root_path, "favicon.png") {
+                    found_favicon_id = "favicon.ico";
+                } else if is_file_in_root(root_path, "favicon.svg") {
+                    found_favicon_id = "favicon.svg";
+                } else if is_file_in_root(root_path, "favicon.png") {
+                    found_favicon_id = "favicon.png";
+                } else if is_file_in_root(root_path, "favicon.jpg") {
+                    found_favicon_id = "favicon.jpg";
+                } else {
+                    // Not using any favicon
+                    return None;
+                }
```

Comment:
```suggestion
                let found_favicon_id = if is_file_in_root(root_path, "favicon.png") {
                    found_favicon_id = "favicon.ico";
                } else if is_file_in_root(root_path, "favicon.svg") {
                    found_favicon_id = "favicon.svg";
                } else if is_file_in_root(root_path, "favicon.png") {
                    found_favicon_id = "favicon.png";
                } else if is_file_in_root(root_path, "favicon.jpg") {
                    found_favicon_id = "favicon.jpg";
                } else {
                    // Not using any favicon
                    return None;
                }
```

Reply:
will this return None to the caller function or get captured by the "found_favicon_id" bcoz I need to return None in case no favicon.* file is found in root.

Reply:
I did a slight modification in your suggestion.